### ISSUE TITLE
Initializing complicated storyboard is too slow

### ISSUE THREAD

**Issue by DCNick3** [ppy/osu#4976]
**Describe the bug:** 

Starting play for [this map](https://osu.ppy.sh/beatmapsets/924551#osu/1949551) with storyboard enabled takes a two and half minutes of one CPU core usage at 100%. This is too long to wait before play or retry. Disabling storyboard makes it load in less than 5 seconds.

OS: Arch Linux
CPU: Intel Core i7-8750H @ 12x 4.1GHz
GPU: GeForce GTX 1060 with Max-Q Design

**osu!lazer version:** 
revision 164b05abd6debb5b817d2f47620e593c006a24b2

**Logs:** 
[runtime.log](https://github.com/ppy/osu/files/3269865/runtime.log)

**Comment by peppy** [comment]
We are very aware of performance issues with storyboard loading. In `osu-stable` loading of storyboards is incremental to offload some of this overhead, but we also have improvements to make in the parsing logic in general.

**Comment by DCNick3** [comment]
Does this mean I shouldn't open issues about poor storyboard performance with specific beatmap?

**Comment by peppy** [comment]
It's not specific to any beatmap. The more storyboard content, the longer it will take.

**Comment by DCNick3** [comment]
Does `Update` and `Draw` drop from 240 and 120 Hz correspondingly to 40 Hz both on [ginkiha - EOS by alacat](https://osu.ppy.sh/beatmapsets/151720#osu/373781) count?

**Comment by peppy** [comment]
See https://github.com/ppy/osu/issues/3756

**Comment by ThatsNiceGuy** [comment]
This issue is very pronounced with [this](https://osu.ppy.sh/beatmapsets/988533#osu/2067820) beatmap with it's storyboard enabled. It's been loading for over 15 minutes and used so much of the machine's memory that it starts swapping to disk sometimes.
tl;dr excessive memory usage upwards of 8gb
![image](https://user-images.githubusercontent.com/10524728/70456709-e4c28e80-1a6b-11ea-95ce-c65a6a05bf2c.png)
lazer 2019.1205.0
Intel Core i7-8750H
GTX 1060 Max-Q
12GB DDR4 @ 2666 MHz

**Comment by Jordan-Jay** [comment]
Storyboards slow loading have been fixed close this issue. The map now takes 14 seconds to load at least on my computer. Freezing also doesn't occur anymore with large storyboard maps.

**Comment by peppy** [comment]
14 seconds is still too long. I would expect this to take 0-5 seconds on a powerful machine.

**Comment by EVAST9919** [comment]
Maps provided in OP and in [this comment](https://github.com/ppy/osu/issues/4976#issuecomment-563338922) are fixed by https://github.com/ppy/osu/pull/27539.

**Comment by peppy** [comment]
> Maps provided in OP and in [this comment](https://github.com/ppy/osu/issues/4976#issuecomment-563338922) are fixed by #27539.

Are there more which remain? May be worth mentioning those specifically else we can close this issue.

**Comment by EVAST9919** [comment]
There's only [this comment](https://github.com/ppy/osu/issues/4976#issuecomment-500481283), but that's another issue.
So I guess this one can be closed.

**Comment by Joehuu** [comment]
There's https://github.com/ppy/osu/issues/11142 and I do get a ~15->5s improvement on an M1 Air w/ 16 GB on master but that's the upper bound of https://github.com/ppy/osu/issues/4976#issuecomment-1304758801.
